### PR TITLE
chore(deps): partially bump bitcoin to v0.32

### PR DESCRIPTION
**Pull request by tvolk131**
Part of #5200

### CODE REVIEW

File: fedimint-core/src/bitcoin_migration.rs [hunk 1]
```diff
@@ -110,6 +110,15 @@ pub fn bitcoin30_to_bitcoin32_payment_preimage(
     lightning_types::payment::PaymentPreimage(preimage.0)
 }
 
+pub fn bitcoin30_to_bitcoin32_sha256_hash(
+    hash: &bitcoin30::hashes::sha256::Hash,
+) -> bitcoin::hashes::sha256::Hash {
+    bincode::deserialize(
+        &bincode::serialize(&hash).expect("Failed to serialize bitcoin30 sha256 hash"),
+    )
+    .expect("Failed to convert bitcoin30 sha256 hash to bitcoin32 sha256 hash")
```
Comment (elsirion) [inline fedimint-core/src/bitcoin_migration.rs:196]: That's the third creative way you have found to transform hashes in rust-bitcoin :laughing: 

When you look at the [`sha256::Hash` impl](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/sha256/struct.Hash.html) there are methods to converting from/to byte array references:

Screenshot 2024-10-24 at 10-22-57 Hash in bitcoin_hashes sha256 - Rust
Screenshot 2024-10-24 at 10-22-49 Hash in bitcoin_hashes sha256 - Rust

**Review by elsirion — CHANGES_REQUESTED** [review]
I'm pretty sure you don't need `bincode` for any of these conversions and we should avoid introducing dependencies whenever posssible. Please look at the struct's docs, there should always be some from/to bytes conversion functions.

File: fedimint-core/src/bitcoin_migration.rs (hunk 1 above)
Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs]: Haha achievement unlocked...?

Fixed. Could probably optimize some of the existing migration functions, but will do that in a follow-up.

Comment (elsirion) [reply fedimint-core/src/bitcoin_migration.rs]: Idk if necssary to touch other fns just yet, ideally we will get rid of them soonish anyway. Just introducing new ones seems very unnecessary. If we get stuck for some reason I'd do the refactor.

Comment (elsirion) [reply fedimint-core/src/bitcoin_migration.rs]: > Fixed

Did you push?

Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs]: > Did you push?

Whoops! Pushed.

File: fedimint-core/src/bitcoin_migration.rs [hunk 2]
```diff
@@ -50,89 +50,74 @@ pub fn checked_address_to_unchecked_address(
...
 
 pub fn bitcoin30_to_bitcoin32_secp256k1_pubkey(
     pubkey: &bitcoin30::secp256k1::PublicKey,
 ) -> bitcoin::secp256k1::PublicKey {
-    bincode::deserialize(
-        &bincode::serialize(&pubkey).expect("Failed to serialize bitcoin30 secp256k1 pubkey"),
-    )
-    .expect("Failed to convert bitcoin30 secp256k1 pubkey to bitcoin32 secp256k1 pubkey")
+    bitcoin::secp256k1::PublicKey::from_str(&pubkey.to_string())
+        .expect("Failed to convert bitcoin30 secp256k1 pubkey to bitcoin32 secp256k1 pubkey")
```
Comment (elsirion) [inline fedimint-core/src/bitcoin_migration.rs:95]: Screenshot 2024-10-27 at 09-01-41 PublicKey in secp256k1 - Rust
Screenshot 2024-10-27 at 09-01-04 PublicKey in secp256k1 - Rust

**Review by elsirion — COMMENTED** [review]
Let's avoid stingification (it's ok for addresses, but keys have better methods for encoding/decoding)

File: fedimint-core/src/bitcoin_migration.rs (hunk 2 above)
Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs]: Done. However, `PublicKey::from_byte_array_compressed()` is introduced in secp256k1 v0.30, but bitcoin v0.30 relies on secp256k1 v0.29, so the best way to do this is using `PublicKey::from_slice()` and then calling `.expect()`.

**Review by elsirion — APPROVED** [review]

**Pull request merged**